File: Ejercicio1/FactoriaMontaña.java
package Ejercicio1;

import java.util.ArrayList;

public class FactoriaMontaña implements FactoriaCarreraYBicicleta {

    @Override
    public Bicicleta crearBicicleta() {
        Bicicleta bici = new BicicletaMontaña();
        return bici;
    }

    @Override
    public Carrera crearCarrera(ArrayList<Bicicleta> bicis) {
        Carrera carrera = new CarreraMontaña(bicis);
        return carrera;
    }

}

File: P1/Ejercicio2/FactoriaMontaña.py
from FactoriaCarreraYBicicleta import FactoriaCarreraYBicicleta
from Carrera import Carrera
from Bicicleta import Bicicleta
from CarreraMontaña import CarreraMontaña
from BicicletaMontaña import BicicletaMontaña
from typing import List

class FactoriaMontaña(FactoriaCarreraYBicicleta):
    def crear_bicicleta(self) -> BicicletaMontaña:
        return BicicletaMontaña()

    def crear_carrera(self, bicis: List[Bicicleta]) -> CarreraMontaña:
        return CarreraMontaña(bicis)